From ccb004ba10cfcdbcc419440e5f877605896bac4c Mon Sep 17 00:00:00 2001
From: Chris Wong <lambda.fairy@gmail.com>
Date: Fri, 14 Oct 2016 22:42:03 +1300
Subject: [PATCH] Update to http-conduit 2.2

The new `parseRequest` function doesn't throw on non-200 responses, so
we don't need to account for these in the exception handler.
---
 hakyll.cabal        | 2 +-
 src/Hakyll/Check.hs | 5 ++---
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/hakyll.cabal b/hakyll.cabal
index 44a09e5..c66524e 100644
--- a/hakyll.cabal
+++ b/hakyll.cabal
@@ -202,7 +202,7 @@ Library
 
   If flag(checkExternal)
     Build-depends:
-      http-conduit >= 2.1    && < 2.2,
+      http-conduit >= 2.2    && < 2.3,
       http-types   >= 0.7    && < 0.10
     Cpp-options:
       -DCHECK_EXTERNAL
diff --git a/src/Hakyll/Check.hs b/src/Hakyll/Check.hs
index c917f3a..b41b40e 100644
--- a/src/Hakyll/Check.hs
+++ b/src/Hakyll/Check.hs
@@ -236,8 +236,7 @@ checkExternalUrl url = do
 
     -- Convert exception to a concise form
     showException e = case cast e of
-        Just (Http.StatusCodeException (Http.Status code msg) _ _) ->
-            show code ++ " " ++ unpack msg
+        Just (Http.HttpExceptionRequest _ e') -> show e'
         _ -> head $ words $ show e
 #else
 checkExternalUrl _ = return ()
